Thinktecture.Relay.Server/Startup.cs

```diff
@@ -90,104 +92,103 @@ private ILifetimeScope RegisterAdditionalServices(ILifetimeScope container, Http
 		{
-			if (app == null)
-				throw new ArgumentNullException(nameof(app));
-			if (authProvider == null)
-				throw new ArgumentNullException(nameof(authProvider));
-
 			// Todo: Add this only when relaying is enabled (no need to auth OnPremises if not), and add
 			// a second endpoint with different token lifetime for management web user (i.e. `/managementToken`),
 			// when management web is enabled. Also, use different AuthProviders for each endpoint
 			var serverOptions = new OAuthAuthorizationServerOptions
 			{
-				AllowInsecureHttp = config.UseInsecureHttp,
+				AllowInsecureHttp = _configuration.UseInsecureHttp,
 				TokenEndpointPath = new PathString("/token"),
-				AccessTokenExpireTimeSpan = config.AccessTokenLifetime,
-				Provider = authProvider,
+				AccessTokenExpireTimeSpan = _configuration.AccessTokenLifetime,
+				Provider = _authorizationServerProvider,
 			};
 
-			var sharedSecret = config.OAuthSharedSecret;
-
-			if (!String.IsNullOrWhiteSpace(sharedSecret))
-			{
-				UseSharedSecret(app, sharedSecret, serverOptions);
-				return;
-			}
-
-			var certBase64 = config.OAuthCertificate;
-			var authOptions = new OAuthBearerAuthenticationOptions();
+			var certBase64 = _configuration.OAuthCertificate;
 
 			if (!String.IsNullOrWhiteSpace(certBase64))
 			{
-				var certRaw = Convert.FromBase64String(certBase64);
-				var oauthCert = new X509Certificate2(certRaw);
+				var cert = new X509Certificate2(Convert.FromBase64String(certBase64));
 
-				serverOptions.AccessTokenFormat = new TicketDataFormat(new RsaDataProtector(oauthCert));
-				authOptions.AccessTokenFormat = new TicketDataFormat(new RsaDataProtector(oauthCert));
-			}
+				serverOptions.AccessTokenFormat = new TicketDataFormat(new RsaDataProtector(cert));
 
-			authOptions.Provider = new OAuthBearerAuthenticationProvider()
-			{
-				OnApplyChallenge = context =>
+				var authOptions = new OAuthBearerAuthenticationOptions
 				{
-					// Workaround: Keep an already set WWW-Authenticate header (otherwise OWIN would add its challenge). 
-					if (!context.Response.Headers.ContainsKey("WWW-Authenticate"))
+					AccessTokenFormat = new TicketDataFormat(new RsaDataProtector(cert)),
+					Provider = new OAuthBearerAuthenticationProvider()
 					{
-						context.OwinContext.Response.Headers.AppendValues("WWW-Authenticate", context.Challenge);
+						OnApplyChallenge = context =>
+						{
+							// Workaround: Keep an already set WWW-Authenticate header (otherwise OWIN would add its challenge). 
+							if (!context.Response.Headers.ContainsKey("WWW-Authenticate"))
+							{
+								context.OwinContext.Response.Headers.AppendValues("WWW-Authenticate", context.Challenge);
+							}
+
+							return Task.CompletedTask;
+						}
 					}
-					return Task.FromResult(0);
-				}
-			};
+				};
 
-			app.UseOAuthBearerAuthentication(authOptions);
-			app.UseOAuthAuthorizationServer(serverOptions);
-		}
+				app.UseOAuthBearerAuthentication(authOptions);
```

Comment:
Why was `app.UseOAuthAuthorizationServer(serverOptions);` removed here? Shouldn't that still be enabled here? Especially as `serverOptions` are prepared beforehand?

Reply:
It's a little bit more down now :)